source/grid/grid_generator.cc

```diff
@@ -28,11 +29,1219 @@
+        {
+          for (auto cell : tria.active_cell_iterators())
+            for (unsigned int f = 0; f < GeometryInfo<2>::faces_per_cell; ++f)
+              {
+                if (cell->face(f)->at_boundary() == false)
+                  continue;
+
+                const auto mid = cell->material_id();
+
+                if ((mid == id_block_1 && f == 0) ||
+                    (mid == id_block_4 && f == 0))
+                  cell->face(f)->set_boundary_id(0); // inlet
+                else if ((mid == id_block_3 && f == 0) ||
+                         (mid == id_block_6 && f == 2))
+                  cell->face(f)->set_boundary_id(1); // outlet
+                else if ((mid == id_block_1 && f == 1) ||
+                         (mid == id_block_2 && f == 1))
+                  cell->face(f)->set_boundary_id(2); // upper airfoil side
+                else if ((mid == id_block_4 && f == 1) ||
+                         (mid == id_block_5 && f == 3))
+                  cell->face(f)->set_boundary_id(3); // lower airfoil side
+                else if ((mid == id_block_2 && f == 0) ||
+                         (mid == id_block_3 && f == 2))
+                  cell->face(f)->set_boundary_id(4); // upper far-field side
+                else if ((mid == id_block_5 && f == 2) ||
+                         (mid == id_block_6 && f == 0))
+                  cell->face(f)->set_boundary_id(5); // lower far-field side
+                else
+                  Assert(false, ExcIndexRange(mid, id_block_1, id_block_6));
+              }
+        }
+
+        /*
+         * Interpolate all vertices of the given triangulation onto the airfoil
+         * geometry, depending on the material_id of the block.
+         * Due to symmetry of coarse grid in respect to
+         * x-axis (by definition of points A-L), blocks 1&4, 2&4 and 3&6 can be
+         * interpolated with the same geometric computations Consider a
+         * bias_factor and incline_factor during interpolation to obtain a more
+         * dense mesh next to airfoil geometry and recieve an inclined boundary
+         * between block 2&3 and 5&6, respectively
+         */
+        void interpolate(Triangulation<2> &tria) const
+        {
+          // array storing the information if a vertex was processed
+          std::vector<bool> vertex_processed(tria.n_vertices(), false);
+
+          // rotation matrix for clockwise rotation of block 1 by angle gamma
+          Tensor<2, 2, double> rotation_matrix_1, rotation_matrix_2;
+
+          rotation_matrix_1[0][0] = +std::cos(-gamma);
+          rotation_matrix_1[0][1] = -std::sin(-gamma);
+          rotation_matrix_1[1][0] = +std::sin(-gamma);
+          rotation_matrix_1[1][1] = +std::cos(-gamma);
+
+          rotation_matrix_2[0][0] = +std::cos(+gamma);
+          rotation_matrix_2[0][1] = -std::sin(+gamma);
+          rotation_matrix_2[1][0] = +std::sin(+gamma);
+          rotation_matrix_2[1][1] = +std::cos(+gamma);
```

Comment:
I *think* that `rotation_matrix_2` is just the transpose of `rotation_matrix_1`, given that `cos` is an even and `sin` an odd function. If so, just write
```
  rotation_matrix_2 = transpose(rotation_matrix_1);
```